ui/src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterfaceFormFields/AddInterfaceFormFields.tsx

```diff
@@ -0,0 +1,134 @@
+import { Col, Input, Row } from "@canonical/react-components";
+import { useFormikContext } from "formik";
+import { useSelector } from "react-redux";
+
+import type { AddInterfaceValues } from "../types";
+
+import FabricSelect from "app/base/components/FabricSelect";
+import FormikField from "app/base/components/FormikField";
+import LinkModeSelect from "app/base/components/LinkModeSelect";
+import MacAddressField from "app/base/components/MacAddressField";
+import SubnetSelect from "app/base/components/SubnetSelect";
+import TagField from "app/base/components/TagField";
+import VLANSelect from "app/base/components/VLANSelect";
+import fabricSelectors from "app/store/fabric/selectors";
+import { NetworkLinkMode } from "app/store/machine/types";
+import subnetSelectors from "app/store/subnet/selectors";
+import type { Subnet } from "app/store/subnet/types";
+
+const fieldOrder = ["fabric", "vlan", "subnet", "mode", "ip_address"];
+
+const AddInterfaceFormFields = (): JSX.Element | null => {
+  const fabrics = useSelector(fabricSelectors.all);
+  const subnets = useSelector(subnetSelectors.all);
+  const { setFieldValue, values } = useFormikContext<AddInterfaceValues>();
+  const setField = <N extends keyof AddInterfaceValues>(
+    name: N,
+    value: AddInterfaceValues[N]
+  ) => {
+    // Update the Formik store with the value.
+    setFieldValue(name, value);
+    // Reset all fields after this one.
+    const position = fieldOrder.indexOf(name);
+    for (let i = position + 1; i < fieldOrder.length; i++) {
+      setFieldValue(fieldOrder[i], "");
+    }
+  };
+  return (
+    <>
+      <Row>
+        <Col size="6">
+          <FormikField label="Name" type="text" name="name" />
+        </Col>
+      </Row>
+      <hr />
+      <Row>
+        <Col size="6">
+          <Input
+            disabled
+            label="Type"
+            value="Physical"
+            type="text"
+            name="type"
+          />
+        </Col>
+        <Col size="6">
+          <FabricSelect
+            name="fabric"
+            onChange={(evt: React.ChangeEvent<HTMLInputElement>) => {
+              const { value } = evt.target;
+              // Manually set the value because we've overwritten the onChange.
+              setFieldValue("fabric", Number(value));
+              if (value) {
```

Comment:
Will this still work if the fabric id is 0?

Reply:
It could be `0` so I'll make this more explicit.